### DISCUSSION TITLE
Instantiating WandBLogger in CLI: assert log_dir is not None

### DISCUSSION THREAD

**Discussion by rbracco** [Lightning-AI/pytorch-lightning#13351] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
I've read the docs and checked issues, but I can't figure out how to instantiate WandBLogger using LightningCLI, as the logger `save_dir` apparently doesn't get passed to the trainer. 

![image](https://user-images.githubusercontent.com/47190785/174830867-cd31d307-76f6-4841-a228-ef70d3ebca9c.png)

results in 
```
new_trainer.py 21 <module>
cli = ExperimentCLI(

cli.py 566 __init__
self._run_subcommand(self.subcommand)

cli.py 837 _run_subcommand
fn(**fn_kwargs)

trainer.py 768 fit
self._call_and_handle_interrupt(

trainer.py 721 _call_and_handle_interrupt
return trainer_fn(*args, **kwargs)

trainer.py 809 _fit_impl
results = self._run(model, ckpt_path=self.ckpt_path)

trainer.py 1172 _run
self._call_setup_hook()  # allow user to setup lightning_module in accelerator environment

trainer.py 1491 _call_setup_hook
self._call_callback_hooks("setup", stage=fn)

trainer.py 1634 _call_callback_hooks
fn(self, self.lightning_module, *args, **kwargs)

cli.py 434 setup
assert log_dir is not None

AssertionError
```

Because the `log_dir` isn't set on the trainer. But if I set it in the config as so...
![image](https://user-images.githubusercontent.com/47190785/174831098-1ba9e76c-f9ff-4648-b763-bda471932c03.png)

it tells me that it isn't being used anywhere.
![image](https://user-images.githubusercontent.com/47190785/174831283-ad1a2762-a472-47ab-9d16-b4d0b460d390.png)

Please help.

**Comment by kubitz** [comment]
Did you solve this issue @rbracco ?

**Reply by rbracco** [reply]
Unfortunately no, I stopped using the cli for the time being but I plan to go back and try again sometime soon. I'll post here if i find a solution

**Reply by tshu-w** [reply]
I have made a PR to fix this issue https://github.com/Lightning-AI/lightning/pull/14878. Currently, you can pass `dir` to `init_args`:
```yaml
init_args:
  save_dir: "logs/"
  dir: "logs/"
```

**Comment by tshu-w** [comment]
I have made a PR to fix this issue https://github.com/Lightning-AI/lightning/pull/14878
